### PR TITLE
Allowing single keys in datastore get/delete.

### LINKED ISSUE #701: datastore.get() requires a list
See http://googlecloudplatform.github.io/gcloud-python/

The snippet is:

```python
from gcloud import datastore
datastore.set_defaults()

product_key = datastore.Key('Product', 123)
print datastore.get([product_key])
```

I'd expect it to be:

```python
from gcloud import datastore
product_key = datastore.Key('Product', 123)
product = datastore.get(product_key)
```

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #701.

### CODE REVIEW

**Comment by tseaver** [comment]
Ugh.  I really hate the fact that the return type varies here based on the argument.

**Comment by dhermes** [comment]
@tseaver Can you hop over into the discussion in #701 and weigh in?

**Comment by tseaver** [comment]
Done

**Comment by dhermes** [comment]
Thanks

File: gcloud/datastore/api.py [hunk 1]
```diff
@@ -191,15 +193,22 @@ def get(keys, missing=None, deferred=None, connection=None, dataset_id=None):
...
     :raises: EnvironmentError if ``connection`` or ``dataset_id`` not passed,
              and cannot be inferred from the environment.  ValueError if
              one or more of ``keys`` has a dataset ID which does not match
              the passed / inferred dataset ID.
     """
-    if not keys:
+    if not key_or_keys:
         return []
 
+    single_key = not isinstance(key_or_keys, list)
```
Comment (jgeewax) [inline gcloud/datastore/api.py]: What about a tuple? or an iterator? Would this be better if we check `single_key = isinstance(key_or_keys, datastore.Key)` ?

Comment (dhermes) [inline gcloud/datastore/api.py]: Moot. I'm throwing this PR based on new discussion in #701 and doing `get`/`get_multi`.

**Pull request closed without merging**